src/main/java/at/ac/tuwien/kr/alpha/api/mapper/impl/AnswerSetToWorkbookMapper.java

```diff
@@ -0,0 +1,140 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package at.ac.tuwien.kr.alpha.api.mapper.impl;
+
+import java.util.List;
+
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellStyle;
+import org.apache.poi.ss.usermodel.FillPatternType;
+import org.apache.poi.ss.usermodel.Font;
+import org.apache.poi.ss.usermodel.IndexedColors;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import at.ac.tuwien.kr.alpha.api.mapper.AnswerSetToObjectMapper;
+import at.ac.tuwien.kr.alpha.common.AnswerSet;
+import at.ac.tuwien.kr.alpha.common.Predicate;
+import at.ac.tuwien.kr.alpha.common.atoms.Atom;
+import at.ac.tuwien.kr.alpha.common.atoms.BasicAtom;
+import at.ac.tuwien.kr.alpha.common.terms.Term;
+
+/**
+ * Implementation of {@link AnswerSetToObjectMapper} that generates an office open xml workbook ("excel file") from a given answer set.
+ * 
+ * Copyright (c) 2020, the Alpha Team.
+ */
+public class AnswerSetToWorkbookMapper implements AnswerSetToObjectMapper<Workbook> {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(AnswerSetToWorkbookMapper.class);
+
+	/**
+	 * Creates an xlsx workbook containing all the atoms from the given {@link AnswerSet} with one sheet per predicate. All predicates with arity 0 are listed
+	 * in a special sheet called "flags". Caution, potential resource leak: note that the returned workbook needs to be closed by the caller once it has been
+	 * processed (written to file etc).
+	 */
+	@Override
+	public Workbook mapFromAnswerSet(AnswerSet answerSet) {
+		LOGGER.debug("Start mapping answer set to workbook");
+		Workbook workbook = new XSSFWorkbook();
+		// create cell style for header cells
+		CellStyle headerStyle = this.createHeaderStyle(workbook);
+
+		// first, create a worksheet for 0-arity predicates
+		Sheet flags = this.createSheetWithHeader(workbook, headerStyle, "Flags", "Flags");
+		Sheet currentPredicateSheet;
+		String[] headerContent;
+		for (Predicate pred : answerSet.getPredicates()) {
+			if (pred.getArity() == 0) {
+				// 0-artiy predicate has no instances in answer set, create a dummy atom
```

Comment:
I am pretty sure that 0-ary atoms do have instances in the answer set, so the creation of the dummy atom should not be necessary.

Reply:
that's right, my mistake... fixed